loader/src/ui/nodes/TextAreaV2.cpp

```diff
@@ -0,0 +1,377 @@
+#include <Geode/ui/TextAreaV2.hpp>
+#include <Geode/DefaultInclude.hpp>
+
+using namespace geode::prelude;
+
+#define IMPL_GETTER(type, name, methodName) \
+    type geode::TextAreaV2::get##methodName() const { return m_##name; }
+#define IMPL_GETTER_SETTER(type, paramType, name, methodName) \
+    void geode::TextAreaV2::set##methodName(const paramType name) { m_##name = name; this->update(); } \
+    IMPL_GETTER(type, name, methodName)
+
+geode::TextAreaV2* geode::TextAreaV2::create(const std::string& text, const std::string& font, const float scale, const float width, const float height, const bool deferUpdates) {
+    TextAreaV2* area = new TextAreaV2(text, font, scale, width, height, deferUpdates);
+
+    if (area && area->init()) {
+        area->autorelease();
+
+        return area;
+    } else {
+        CC_SAFE_DELETE(area);
+
+        return nullptr;
+    }
+}
+
+geode::TextAreaV2::TextAreaV2(const std::string& text, const std::string& font, const float scale, const float width, const float height, const bool deferUpdates) :
+    PaddingNode(CCNode::create()),
+    m_text(text),
+    m_font(font),
+    m_textColor({ 255, 255, 255, 255 }),
+    m_alignment(Alignment::Left),
+    m_wrappingMode(WrappingMode::WordWrap),
+    m_maxLines(-1),
+    m_textScale(scale),
+    m_linePadding(0),
+    m_minWidth(-1),
```

Comment:
If a value is optional and its null state isn't logically 0, it should be explicitly marked `std::optional`

Reply:
Aka the member should be `std::optional` and this should be `std::nullopt`

Reply:
I opted to not do this since there's frequent math usage involved with these making it hell to use if they're optional